CluedoSWEDocs/man/man3/MenuUI.3
.TH "MenuUI" 3 "CluedoSWE" \" -*- nroff -*-
.ad l
.nh
.SH NAME
MenuUI \- Class \fR\fBMenuUI\fP\fP represents the highest level of the menu UI, containing attributes to be used across the main menu\&.  

.SH SYNOPSIS
.br
.PP
.PP
Inherits MonoBehaviour\&.
.SS "Public Member Functions"

.in +1c
.ti -1c
.RI "int \fBgetPlayerNum\fP ()"
.br
.RI "Returns the number of players to be used ingame\&. "
.ti -1c
.RI "Sprite \fBGetBackgroundSprite\fP (int spriteNum)"
.br
.RI "Returns the background sprite corresponding to the int passed in\&. "
.ti -1c
.RI "void \fBSetBackground\fP (int spriteNum)"
.br
.ti -1c
.RI "Sprite \fBGetToggleSprite\fP (int spriteNum)"
.br
.RI "Returns the toggle sprite corresponding to the int passed in\&. "
.ti -1c
.RI "void \fBSetToggleSprite\fP (int spriteNum)"
.br
.ti -1c
.RI "void \fBLoadCards\fP (int cardNum)"
.br
.in -1c
.SS "Static Public Member Functions"

.in +1c
.ti -1c
.RI "static void \fBToggleElement\fP (GameObject element, bool state)"
.br
.in -1c
.SH "Detailed Description"
.PP 
Class \fR\fBMenuUI\fP\fP represents the highest level of the menu UI, containing attributes to be used across the main menu\&. 
.SH "Member Function Documentation"
.PP 
.SS "Sprite MenuUI\&.GetBackgroundSprite (int spriteNum)"

.PP
Returns the background sprite corresponding to the int passed in\&. 
.PP
\fBParameters\fP
.RS 4
\fIspriteNum\fP An int representing the background sprite to be returned\&.
.RE
.PP
\fBReturns\fP
.RS 4
The background sprite corresponding to the int passed in\&.
.RE
.PP

.SS "int MenuUI\&.getPlayerNum ()"

.PP
Returns the number of players to be used ingame\&. 
.PP
\fBReturns\fP
.RS 4
The number of players to be used ingame\&.
.RE
.PP

.SS "Sprite MenuUI\&.GetToggleSprite (int spriteNum)"

.PP
Returns the toggle sprite corresponding to the int passed in\&. 
.PP
\fBParameters\fP
.RS 4
\fIspriteNum\fP An int representing the toggle sprite to be returned\&.
.RE
.PP
\fBReturns\fP
.RS 4
The toggle sprite corresponding to the int passed in\&.
.RE
.PP


.SH "Author"
.PP 
Generated automatically by Doxygen for CluedoSWE from the source code\&.


Assets/GameAssets/Scripts/MainMenu/MenuUI.cs
using UnityEditor;
using UnityEngine;
using UnityEngine.UI;

/// <summary>
/// Class <c>MenuUI</c> represents the highest level of the menu UI, containing attributes to be used across the main menu.
/// </summary>
[RequireComponent(typeof(Image))]
public class MenuUI : MonoBehaviour
{
	private int playerNum = 0;
    [SerializeField] Sprite[] backgroundSprites;
    [SerializeField] Sprite[] toggleSprites;
	[SerializeField] Sprite[] cardSprites;

    // Start is called once before the first execution of Update after the MonoBehaviour is created
    void Start()
    {
		GameObject cardPanel = transform.GetChild(3).gameObject.transform.GetChild(1).gameObject;
		ResetCards(cardPanel);
		ResetToggles(cardPanel);
		ToggleElement(transform.GetChild(3).transform.GetChild(0).transform.GetChild(7).gameObject, false);
	}

    // Update is called once per frame
    void Update()
    {
        
    }

	/// <summary>
	/// Returns the number of players to be used ingame.
	/// </summary>
	/// <returns>The number of players to be used ingame.</returns>
	public int getPlayerNum()
	{
		return playerNum;
	}

    /// <summary>
    /// Returns the background sprite corresponding to the int passed in.
    /// </summary>
    /// <param name="spriteNum">An int representing the background sprite to be returned.</param>
    /// <returns>The background sprite corresponding to the int passed in.</returns>
    public Sprite GetBackgroundSprite(int spriteNum)
    {
        return backgroundSprites[spriteNum];
    }

	public void SetBackground(int spriteNum)
	{
		Sprite currentBackgroundSprite = GetComponent<Image>().sprite;
		Sprite newBackgroundSprite = backgroundSprites[spriteNum];
		GetComponent<Image>().sprite = newBackgroundSprite;
	}

	/// <summary>
	/// Returns the toggle sprite corresponding to the int passed in.
	/// </summary>
	/// <param name="spriteNum">An int representing the toggle sprite to be returned.</param>
	/// <returns>The toggle sprite corresponding to the int passed in.</returns>
	public Sprite GetToggleSprite(int spriteNum)
	{
		return toggleSprites[spriteNum];
	}

	/// <summary>
	/// Loads the correct number of cards onto the menu screen.
	/// </summary>
	/// <param name="cardNum">Number of cards to load.</param>
    public void LoadCards(int cardNum)
    {
		GameObject cardPanel = transform.GetChild(3).gameObject.transform.GetChild(1).gameObject;
		ResetCards(cardPanel);
		for (int i = 0; i < cardNum; i++)
		{
            GameObject cardCanvas = cardPanel.transform.GetChild(i).gameObject;
			ToggleElement(cardCanvas, true);
			cardCanvas.transform.GetChild(1).GetComponent<Image>().sprite = cardSprites[i];
		}
		playerNum = cardNum;
	}

	/// <summary>
	/// Removes all the cards from the menu screen.
	/// </summary>
	/// <param name="cardPanel">The cardPanel to remove cards from.</param>
	private void ResetCards(GameObject cardPanel)
	{
		for (int i = 0; i < 6; i++)
		{
			GameObject cardCanvas = cardPanel.transform.GetChild(i).gameObject;
			ToggleElement(cardCanvas, false);
			cardCanvas.transform.GetChild(1).GetComponent<Image>().sprite = null;
		}
		playerNum = 0;
	}

	/// <summary>
	/// Toggles whether an element is visible or not.
	/// </summary>
	/// <param name="element">The element to toggle.</param>
	/// <param name="state">The state to set the element to.</param>
	public static void ToggleElement(GameObject element, bool state)
	{
		element.GetComponent<CanvasGroup>().alpha = state ? 1 : 0;
		element.GetComponent<CanvasGroup>().blocksRaycasts = state;
		element.GetComponent<CanvasGroup>().interactable = state;
	}

	/// <summary>
	/// Resets all of the card toggles to CPU (default).
	/// </summary>
	/// <param name="cardPanel">The card panel containing the card toggles.</param>
	private void ResetToggles(GameObject cardPanel)
	{
		for (int i = 0; i < 6; i++)
		{
			GameObject cardCanvas = cardPanel.transform.GetChild(i).gameObject;
			cardCanvas.transform.GetChild(0).GetComponent<Image>().sprite = toggleSprites[0];
		}
	}
}
